### PR TITLE
Initial surfacing of results and facets in the UI

### LINKED ISSUE #21: Results Extraction
Success Criteria

- Convert TIMDEX response into Facets, Records, Pagination blocks
- Display extracted information in UI

Out of scope

- Error handling (we'll handle that in Analyzers maybe?)

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This refactors the QueryBuilder and TimdexWrapper for cleaner application flow (first commit) and then surfaces the returned results and facets in the UI (second commit).

This should close #21 when it merges.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

YES - adding byebug for debugging in test/dev

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -20,6 +20,7 @@ group :production do
 end
 
 group :development, :test do
+  gem 'byebug'
```
Comment (JPrevost) [inline Gemfile:23]: Let's pull out `byebug`. Rails 7 now replaces with this `debugger` that acts the same way without an extra gem.

File: test/controllers/basic_search_controller_test.rb [hunk 2]
```diff
@@ -27,52 +27,89 @@ class BasicSearchControllerTest < ActionDispatch::IntegrationTest
...
+      assert_select '#results'
+      assert_select '#results ul li', { minimum: 1 }
+    end
+  end
 
-    assert_select '#results'
+  test 'searches with zero results are handled gracefully' do
+    VCR.use_cassette('timdex no results',
+                     allow_playback_repeats: true) do
+      get '/results?q=asdfiouwenlasd'
```
Comment (JPrevost) [inline test/controllers/basic_search_controller_test.rb:103]: Note to self, publish book called `asdfiouwenlasd` to break this test in the future.

**Review by JPrevost — CHANGES_REQUESTED** [review]
Will be 👍🏻 once you remove `byebug`.

https://edgeguides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem

File: Gemfile (hunk 1 above)
Comment (matt-bernhardt) [reply Gemfile:23]: oh, cool - sure thing

**Review by JPrevost — APPROVED** [review]

**Pull request merged**